catalog/controller/payment/securesubmit.php

```diff
@@ -40,80 +144,305 @@ public function index() {
 		if (file_exists(DIR_TEMPLATE . $this->config->get('config_template') . '/template/payment/securesubmit.tpl')) {
 			return $this->load->view($this->config->get('config_template') . '/template/payment/securesubmit.tpl', $data);
 		} else {
-			return $this->load->view('default/template/payment/securesubmit.tpl', $data);
+			return $this->load->view('payment/securesubmit.tpl', $data);
 		}
 	}
 
-	public function send() {
+    /**
+     *Process payment
+     */
+    public function send() {
 		$this->load->model('checkout/order');
+        $orderID = $this->session->data['order_id'];
+        // \ModelCheckoutOrder::getOrder
+        $order_info = $this->model_checkout_order->getOrder($orderID);
+        $curCode = $order_info['currency_code'];
+        $amt = $order_info['total'];
+        $json = array();
 
-		$order_info = $this->model_checkout_order->getOrder($this->session->data['order_id']);
+        // \ControllerPaymentSecureSubmit::getHpsCardHolder
+        // builds the \HpsAddress as well
+        // do this here since it is not specific only to credit card transactions
+        $cardHolder = $this->getHpsCardHolder($order_info);
 
-		require_once(DIR_APPLICATION . 'controller/payment/securesubmitfiles/Hps.php');
+		try {
+            /*
+             * get \HpsAuthorization
+             * in order to facilitate this we will retrieve the setting determining which call to make
+             */
+            $response = $this->chargeCreditCard($amt, $curCode, $cardHolder);
+            $json['redirect'] = $this->buildCreditSuccessMessage($orderID, $response);
+		}
+        catch (HpsException $e) {
+            $maxFail = $this->securesubmit_fraud_fail;
 
-        $json = array();
+            if ($this->velocityIncrement() < $maxFail){
+                $this->sessionVar('previous',$e->getMessage());
+            }
+            $json['error'] = $e->getMessage();
+        }
+        catch (Exception $e) {
+			$json['error'] = $e->getMessage();
+		}
```

Comment:
@ttnnkkrr Would you mind cleaning up the tabs in this file (I know it wasn't perfect to begin with)? This method in particular is pretty bad. It's a bit hard to read as is.